guides/html-tag-processor.md

```diff
@@ -0,0 +1,235 @@
+# Using the HTML Tag Processor 
+
+## Reference Guide WordPress Documentation:
+* [Introducing the HTML API in WordPress 6.2](https://make.wordpress.org/core/2023/03/07/introducing-the-html-api-in-wordpress-6-2/)
+
+### Other Guides on the Topic
+* [How to Modify HTML in a PHP WordPress Plugin Using The New Tag Processor API](https://adamadam.blog/2023/02/16/how-to-modify-html-in-a-php-wordpress-plugin-using-the-new-tag-processor-api/)
+* [Changing HTML Tag Attributes with the WP HTML Tag Processor](https://wpdevelopment.courses/articles/wp-html-tag-processor/)
+
+## Use Case:
+For static blocks we can use the registerBlockExtension to easily add additional classnames. Technically it also works for adding inline styles but we should use that sparingly because it may introduce block validation issues and deprecations.
+
+Another issue is that the block extensions only apply to static blocks. When the rendering happens on the server in PHP it doesn't do anything.
+
+To solve for all these usecases we can use the HTMLTagProcessor and the block render callbacks.
+
+### Caveats:
+* This was only introduced in WP 6.2 earlier versions do not have this capability
+
+## Code Example:
+This example showcases how to use `WP_HTML_Tag_Processor` with new block attributes. This example extends the `core/button` block and adds in fields for `Accessible Text` and `Button Padding` these new fields can then be used within PHP to create filters with the `WP_HTML_Tag_Processor` to update values of the rendered block content. In this case add in an `aria-label` with the text provided and set the `padding` around the button with the value provided via the block settings.
+
+`block-filters/button.js`
+```bash
```

Comment:
Docusaurus code blocks allow us to specify a title right inline. Also we should use the correct syntax highlighter language 

```suggestion
```js title="block-filters/button.js"
```